### PR TITLE
#162053826 Implement on-boarding email automation 

### PR DESCRIPTION
**Pull request by Collins33**
#### What does this PR do?
Add onboarding email automation to the background worker

#### Description of Task to be completed?
- Create onboarding email file and add the onboarding function
- The function sends an email to the developer with the partner engagement guide.
- The function sends email to success-ops with info about the developer who has been placed.

#### How should this be manually tested?
1. Set environment variable: `TIMER_INTERVAL=15s`
2. Log `automationResult` object for each placement in `executeJobs` function
2. On the terminal, run: `npm start` or `yarn start`
3. Watch the terminal to view the status of email onboarding jobs, printed every 15 seconds interval

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#162053826

#### Postman Documentation
N/A

#### Screenshots (If applicable)
N/A

### CODE REVIEW

File: server/jobs/onboarding/email.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @param {array} placement Placement record whose developer is to be offboarded
+ * @param {object} automationResult Result of automation job
+ * @returns {void}
+ */
+const emailOnboarding = async (placement, automationResult) => {
+  const { fellow } = placement;
+  const { client_name: partnerName } = placement;
+  const { client_location: partnerLocation } = placement;
+  const { start_date: startDate } = placement;
```
Comment (IAMOTZ) [inline server/jobs/onboarding/email.js]: You can combine all the destructuring above into a single statement like this:
```
const {
    fellow, client_name: partnerName, client_location: partnerLocation, start_date: startDate,
  } = placement;
```

File: server/jobs/onboarding/email.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  try {
+    await developerEmailTransport(fellow.email, fellow.name, partnerName);
+    // eslint-disable-next-line max-len
+    await opsEmailTransport(fellow.name, partnerName, fellow.email, fellow.location, partnerLocation, startDate);
```
Comment (IAMOTZ) [inline server/jobs/onboarding/email.js]: `partnerLocation` is not available in the `placement` object from allocations, you can hardcode a demo value(eg `Kenya`) for now.

File: server/jobs/onboarding/email.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    automationResult.slackOnBoarding = 'success';
+  } catch (error) {
+    automationResult.slackOnBoarding = 'failure';
```
Comment (IAMOTZ) [inline server/jobs/onboarding/email.js]: I guess you forgot to change something here, this isn't `slackOnBoarding`.

File: server/jobs/onboarding/email.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-param-reassign */
+import { developerEmailTransport, opsEmailTransport } from '../../modules/email/emailModule';
+/**
+ * @desc Automates developer onboarding on email
+ *
+ * @param {array} placement Placement record whose developer is to be offboarded
+ * @param {object} automationResult Result of automation job
+ * @returns {void}
```
Comment (IAMOTZ) [inline server/jobs/onboarding/email.js]: Please update the JSDoc here. `placement` is an object and not an `array`.
The function also returns `undefined`(not returning anything) and not `void`.

File: server/jobs/onboarding/email.js (hunk 1 above)
Comment (Collins33) [reply server/jobs/onboarding/email.js]: I have implemented it

File: server/jobs/onboarding/email.js (hunk 2 above)
Comment (Collins33) [reply server/jobs/onboarding/email.js]: I have replaced it with Kenya

File: server/jobs/onboarding/email.js (hunk 3 above)
Comment (Collins33) [reply server/jobs/onboarding/email.js]: I have resolved it

File: server/jobs/onboarding/email.js (hunk 4 above)
Comment (Collins33) [reply server/jobs/onboarding/email.js]: I have resolved it

**Pull request merged**